src/analysis/timeline/timelineCredView.test.js

```diff
@@ -0,0 +1,142 @@
+    const endTimeMs = +new Date(2017, 6);
+    const boundaries = utcWeek.range(startTimeMs, endTimeMs);
+    const intervals = [];
+    for (let i = 0; i < boundaries.length - 1; i++) {
+      intervals.push({
+        startTimeMs: +boundaries[i],
+        endTimeMs: +boundaries[i + 1],
+      });
+    }
+    const users = [
+      ["starter", (x) => Math.max(0, 20 - x)],
+      ["steady", (_) => 4],
+      ["finisher", (x) => (x * x) / 20],
+      ["latecomer", (x) => Math.max(0, x - 20)],
+    ];
+
+    const graph = new Graph();
+    const addressToCred = new Map();
+    for (const [name, generator] of users) {
+      const address = NodeAddress.fromParts(["foo", name]);
+      graph.addNode({
+        address,
+        description: `[@${name}](https://github.com/${name})`,
+        timestampMs: null,
+      });
+      const scores = intervals.map((_unused_, i) => generator(i));
+      addressToCred.set(address, scores);
+    }
+    const filteredTimelineCred: FilteredTimelineCred = {
+      intervals,
+      addressToCred,
+    };
+    const params = {alpha: 0.05, intervalDecay: 0.5, weights: defaultWeights()};
+    return new TimelineCred(graph, filteredTimelineCred, params, credConfig());
+  }
+
+  it("JSON serialization works", () => {
+    const json = exampleTimelineCred().toJSON();
+    const tc_ = TimelineCred.fromJSON(json, credConfig());
+    const tcv = new TimelineCredView(exampleTimelineCred());
+    const tcv_ = new TimelineCredView(tc_);
+    expect(tcv.credSortedNodes(NodeAddress.empty)).toEqual(
+      tcv_.credSortedNodes(NodeAddress.empty)
+    );
+  });
+
+  it("default interval works", () => {
+    const tc = exampleTimelineCred();
+    const tcv = new TimelineCredView(tc);
+    expect(tc.intervals()).toEqual(tcv.intervals());
+  });
+
+  it("single input interval works", () => {
+    const tc = exampleTimelineCred();
+    const startTimeMs = +new Date(2017, 1);
+    const inputInterval = {startTimeMs: startTimeMs, endTimeMs: startTimeMs};
+    const tcv = new TimelineCredView(tc, inputInterval);
+    expect(tcv.intervals().length).toEqual(1);
+  });
+
+  it("straddle input interval works", () => {
+    const tc = exampleTimelineCred();
+    const startTimeMs = +new Date(2017, 0, 14);
+    const endTimeMs = +new Date(2017, 0, 14 + 7 + 2);
+    const inputInterval = {startTimeMs: startTimeMs, endTimeMs: endTimeMs};
+    const tcv = new TimelineCredView(tc, inputInterval);
+    expect(tcv.intervals().length).toEqual(3);
+  });
+
+  it("cred sorting works", () => {
+    const tcv = new TimelineCredView(exampleTimelineCred());
+    const sorted = tcv.credSortedNodes(NodeAddress.empty);
+    const expected = sortBy(sorted, (x) => -x.total);
+    expect(sorted).toEqual(expected);
+  });
+
+  it("cred interval consistency checks", () => {
+    const boundaries = utcWeek.range(+new Date(2017, 0), +new Date(2017, 6));
+    const startTimeIndex = Math.floor(Math.random() * boundaries.length - 1);
```

Comment:
Using `Math.random()` in test code is a bad practice; it invites flakey test failures.

For example, suppose that there are 60 possible configurations of startTimeIndex and endTimeIndex using the Math.random testing style, and only 1 of them is broken. Then in expectation, this commit will merge without test failure, and a week or two later we'll first notice the flakey test failure.